### PR TITLE
Refactor: use `database_transactional_test!()` macro

### PR DESCRIPTION
**Pull request by s373r**
### Changed
- All tests related to databases use the `database_transactional_test` macro
- Some skipped tests will now also be run
- Access token with duplicate names can be created if such name exists but was revoked (now for MySQL as well)

## Checklist before requesting a review ✅ missed tests have been added

- [x] Unit and integration tests added
  
- [x] Compatibility:
    
  - [x] Network APIs: ✅
    
  - [x] Workspace layout and metadata: ✅ 
    
  - [x] Configuration: ✅
    
  - [x] Container images: ✅
- [x] Documentation:
    
  - [x] [Changelog](./CHANGELOG.md): ✅
    
  - [x] [Public documentation](https://github.com/kamu-data/kamu-docs/): ❌ not needed
  
- [x] Downstream effects:
    
  - [x] [kamu-node](https://github.com/kamu-data/kamu-node): ❌ not needed

### CODE REVIEW

File: migrations/mysql/20240823165328_modify_access_token_uniq.sql [hunk 1]
```diff
@@ -0,0 +1,5 @@
+ALTER TABLE access_tokens
+    ADD revoked_at_is_null BOOLEAN GENERATED ALWAYS AS (revoked_at IS NULL);
+
+CREATE OR REPLACE UNIQUE INDEX idx_account_token_name
+    ON access_tokens (account_id, token_name, revoked_at_is_null);
```
Comment (s373r) [inline migrations/mysql/20240823165328_modify_access_token_uniq.sql:5]: 1) To fix the previously missed test
2) MariaDB does not support indexes with predicates

File: src/infra/accounts/repo-tests/src/access_token_repository_test_suite.rs [hunk 2]
```diff
@@ -114,7 +114,7 @@ pub async fn test_insert_and_locate_multiple_access_tokens(catalog: &Catalog) {
 
 ////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////
 
-pub async fn test_mark_existing_access_token_revorked(catalog: &Catalog) {
+pub async fn test_mark_existing_access_token_revoked(catalog: &Catalog) {
```
Comment (s373r) [inline src/infra/accounts/repo-tests/src/access_token_repository_test_suite.rs:117]: Typo: revorked -> revoked

File: src/infra/accounts/repo-tests/src/access_token_repository_test_suite.rs [hunk 3]
```diff
@@ -195,7 +195,7 @@ pub async fn test_create_duplicate_active_access_token(catalog: &Catalog) {
 
 ////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////
 
-pub async fn test_create_duplicate_access_token_err(catalog: &Catalog) {
+pub async fn test_create_duplicate_access_token_error(catalog: &Catalog) {
```
Comment (s373r) [inline src/infra/accounts/repo-tests/src/access_token_repository_test_suite.rs:198]: Expand err -> error

File: src/infra/accounts/repo-tests/src/access_token_repository_test_suite.rs [hunk 4]
```diff
@@ -230,7 +230,7 @@ pub async fn test_create_duplicate_access_token_err(catalog: &Catalog) {
 
 ////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////
 
-pub async fn test_mark_non_existing_access_token_revorked(catalog: &Catalog) {
+pub async fn test_mark_non_existing_access_token_revoked(catalog: &Catalog) {
```
Comment (s373r) [inline src/infra/accounts/repo-tests/src/access_token_repository_test_suite.rs:233]: Typo: revorked -> revoked

File: src/infra/flow-system/inmem/tests/tests/test_inmem_flow_event_store.rs [hunk 5]
```diff
@@ -21,7 +21,7 @@ use opendatafabric::{AccountID, DatasetID};
 ////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////
 
 #[test_log::test(tokio::test)]
-async fn test_dataset_flow_empty_filters_distingush_dataset() {
+async fn test_dataset_flow_empty_filters_distinguish_dataset() {
```
Comment (s373r) [inline src/infra/flow-system/inmem/tests/tests/test_inmem_flow_event_store.rs:24]: Typo: distingush -> distinguish

**Review by zaychenko-sergei — APPROVED** [review]

**Pull request merged**